File: IT/Eccezioni/Esercizi/Python/esercizio10.py
# file esercizio10.py

from __future__ import print_function
if hasattr(__builtins__, 'raw_input'): input = raw_input
from math import sqrt

sq = -1

while (sq < 0):
	s = input("Scrivi un numero: ")

	try:
		n = int(s)
		sq = sqrt(n)
		print("Radice quadrata di {} = {}".format(n, sq))

	except ValueError as e:
		print(e)

print("Fine del programma.")

File: IT/Eccezioni/Esercizi/Java/Esercizio10.java
/* file Esercizio10.java */

public class Esercizio10 {
	public static void main(String [] args) {
		double sq = -1;

		while (sq < 0) {
			String s = System.console().readLine("Scrivi un numero: ");

			try {
				int n = Integer.parseInt(s);

				if (n < 0) {
					throw new IllegalArgumentException("Errore di dominio matematico: radice quadrata indefinita per i numeri negativi");
				}

				sq = Math.sqrt(n);
				System.out.println("Radice quadrata di " + n + " = " + sq);
			}
			catch (NumberFormatException e) {
				System.out.println(e.getMessage());
			}
			catch (IllegalArgumentException e) {
				System.out.println(e.getMessage());
			}
		}

		System.out.println("Fine del programma.");
	}
}

